src/Npgsql/NpgsqlDataSource.cs

```diff
@@ -124,8 +124,9 @@ internal NpgsqlDataSource(
 
             _timerPasswordProviderCancellationTokenSource = new();
 
-            // Create the timer, but don't start it; the manual run below will will schedule the first refresh.
-            _periodicPasswordProviderTimer = new Timer(state => _ = RefreshPassword(), null, Timeout.InfiniteTimeSpan, Timeout.InfiniteTimeSpan);
+            // Create the timer, but don't start it; the manual run below will schedule the first refresh.
+            using (ExecutionContext.SuppressFlow()) // Don't capture the current ExecutionContext and its AsyncLocals onto the timer causing them to live forever
```

Comment:
We might want to extend that so the first refresh attempt behaves the exact same way as the later ones.